src/Datasource/Paginator.php

```diff
@@ -401,6 +406,35 @@ public function validateSort(RepositoryInterface $object, array $options)
         return $options;
     }
 
+    /**
+     * Remove alias if needed.
+     *
+     * @param array $fields Current fields
+     * @param string $model Current model alias
+     * @return array $fields Unaliased fields where applicable
+     */
+    protected function _removeAliases($fields, $model)
+    {
+        $result = [];
+        foreach ($fields as $field => $sort) {
+            if (strpos($field, '.') === false) {
+                $result[$field] = $sort;
+                continue;
+            }
+
+            list ($alias, $currentField) = explode('.', $field);
+
+            if ($alias !== $model) {
```

Comment:
Why is this version `!==` but the one in the helper uses `===`?

Reply:
This was just a bail early here, I can invert the logic to be in sync.